src/helpers/charts/chart_common.ts

```diff
@@ -182,11 +182,6 @@ export function createDataSets(
           )
         );
       }
-    } else if (zone.left === zone.right && zone.top === zone.bottom) {
```

Comment:
In the end, all the diff in this PR except this one is useless right ? (and the tests)
Could you revert the useless diff ?
Even if some changes are clearer, it's not worth it here IMO (especially for a bug fix in stable). Useless changes makes the review more difficult, forward-ports (and future forward-ports) more likely to conflict.

As a general rule of thumb, keep the diff minimal for bug fixes in stable :wink:  Refactorings are better in master and in different commits (not burried in a bug fix or a new feature)

Reply:
Yes, you're right. I though the other diff with the `this.dataSetsHaveTitle` things were necessary, but not with this change :)